src/snowflake/connector/result_set.py

```diff
@@ -180,9 +181,24 @@ def _fetch_pandas_batches(self, **kwargs) -> Iterator[pandas.DataFrame]:
         self._can_create_arrow_iter()
         return self._create_iter(iter_unit=IterUnit.TABLE_UNIT, structure="pandas")
 
+    def _convert_string_to_json(self, dataframe: pandas.DataFrame) -> pandas.DataFrame:
+        """Converts string columns to JSON columns.
+
+        This is necessary for the JSON columns in the dataframe to be
+        serialized properly.
+        """
+        for col in dataframe:
+            if dataframe[col].dtype == "object":
+                try:
+                    dataframe[col] = dataframe[col].apply(loads)
+                    logger.debug(f"Converted column {col} to json")
+                except Exception as e:
```

Comment:
Do we expect some specific error? Using `Exception` is generally discouraged if possible

Reply:
I see one of these two:

the JSON object must be str, bytes or bytearray, not dict
TypeError

or
Expecting value: line 1 column 1 (char 0)
JSONDecodeError

The first one occurs if the data has already been converted to json somehow. I think I only see it because I have the modified connector installed.

The JSONDecodeError is any type that gets converted to PyObject like timestamps etc that aren't json.

So I could `execpt JSONDecodeError as e`
and that would probably do it. If there were a lib that inspects if a string is convertible to JSON, i could do that instead of the try/except.

Reply:
fixed